### PR TITLE
List printer

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PrinterRestController.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+  };
+
+  @Autowired
+  private PrinterService printerService;
+
+  @RequestMapping(method = RequestMethod.POST, headers = { "Content-type=application/json" })
+  @ResponseBody
+  public PrinterDto create(@RequestBody PrinterDto dto) throws IOException {
```
Comment (djcooke) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PrinterRestController.java]: `@ResponseStatus(HttpStatus.CREATED)`

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PrinterRestController.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  @RequestMapping(value = "dt", method = RequestMethod.GET, produces = "application/json")
+  @ResponseBody
+  public DataTablesResponseDto<PrinterDto> dataTable(HttpServletRequest request,
+      HttpServletResponse response, UriComponentsBuilder uriBuilder) throws IOException {
+    return jQueryBackend.get(request, response, uriBuilder);
+  }
+
+  @RequestMapping(method = RequestMethod.DELETE, headers = { "Content-type=application/json" })
+  @ResponseBody
+  public ResponseEntity<?> delete(@RequestBody List<Long> printerIds) throws IOException {
```
Comment (djcooke) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PrinterRestController.java]: `@ResponseStatus(HttpStatus.NO_RESPONSE)`

Then you can change return type to void and remove the return statement

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PrinterRestController.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      if (printer != null) {
+        printerService.remove(printer);
+      }
+    }
+    return new ResponseEntity<>(HttpStatus.OK);
+  }
+
+  @RequestMapping(value = "/disable", method = RequestMethod.PUT, headers = { "Content-type=application/json" })
+  @ResponseBody
+  public ResponseEntity<?> disable(@RequestBody List<Long> printerIds) throws IOException {
```
Comment (djcooke) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PrinterRestController.java]: `@ResponseStatus(HttpStatus.NO_RESPONSE)`, void return type

Same for `/enable`

**Comment by djcooke** [comment]
Could also use `ResponseStatus.OK` where I suggested `NO_CONTENT`, but no content is more accurate, and plays better with newer jQuery ajax methods that I think we mostly don't use for some reason

**Comment by apmasell** [comment]
Okay. I've switched the response types.

**Pull request merged**